execution_engine/src/core/engine_state/mod.rs

```diff
@@ -1798,4 +1803,54 @@ where
             commit_result => Ok(commit_result),
         }
     }
+
+    /// Obtains validator weights for given era.
+    pub fn get_era_validators(
+        &self,
+        correlation_id: CorrelationId,
+        get_era_validators_request: GetEraValidatorsRequest,
+    ) -> Result<GetEraValidatorsResult, Error> {
+        let protocol_version = get_era_validators_request.protocol_version();
+
+        let tracking_copy = match self.tracking_copy(get_era_validators_request.state_hash())? {
+            Some(tracking_copy) => Rc::new(RefCell::new(tracking_copy)),
+            None => return Ok(GetEraValidatorsResult::RootNotFound),
+        };
+
+        let tracking_copy = tracking_copy.borrow();
+
+        let protocol_data = match self.get_protocol_data(protocol_version) {
+            Ok(Some(protocol_data)) => protocol_data,
+            Ok(None) => return Err(Error::InvalidProtocolVersion(protocol_version)),
+            Err(err) => return Err(err),
+        };
+
+        let query_key = Key::Hash(protocol_data.auction());
+
+        let result = tracking_copy
+            .query(correlation_id, query_key, &[ERA_VALIDATORS_KEY.to_string()])
```

Comment:
I understand why you are doing this via query on a tracking copy, but I was under the strong presumption that this would be done via an actual call to the auction contract to get its state. While this technically works I am of the opinion that this breaks the abstraction of the auction contract's api. I think it would be more correct to add an endpoint to the auction contract to return the era validators and call that from this method.

Reply:
Done